docs/features/tuples.md

```diff
@@ -0,0 +1,227 @@
+
+A lot of details and motivation for the feature is given in 
+[https://github.com/dotnet/roslyn/issues/347](https://github.com/dotnet/roslyn/issues/347)
+
+Current design notes:
+https://github.com/dotnet/roslyn/issues/10429
+
+NOTE: The goal of this document is capture what is being implemented. As design evolves, the document will undergo adjustments. 
```

Comment:
It will gradually become one. I did not have time to make it more spec-like

It does reflect what is in the branch though. We may have alternative branches - like investigating "covariant" tuple convertibility. Docs in those branches will different.